openvdb/tree/Tree.h

```diff
@@ -1487,17 +1487,17 @@ template<typename RootNodeType>
 inline void
 Tree<RootNodeType>::clear()
 {
-    std::vector<LeafNodeType*> leafnodes;
```

Comment:
Rather than trying to change the behavior of `stealNodes`, can't you just reimplement `Tree::clear` using `NodeManager`?

@kmuseth Is the current implementation even correct?  Why does it steal only leaf nodes and first-level internal nodes and not second-level internal nodes?  Would it work for trees with more than four levels?

Reply:
That's a good point., a NodeManager::foreachBottomUp() should work for this.

Reply:
I had a go at this and would quite like to hear your thoughts Peter?

Reply:
That looks pretty good.  I'm not sure the `const_cast` is necessary, since `getChild` is const, but it might be safer to leave it in place and do something like this:
```
auto& item = children[iter.pos()];
delete item.getChild();
item.setChild(nullptr);
```
I envisioned the leaf level being a no-op; is there a measurable gain from swapping out the buffer?

Reply:
> That looks pretty good. I'm not sure the `const_cast` is necessary, since `getChild` is const, but it might be safer to leave it in place and do something like this:
> 
> ```
> auto& item = children[iter.pos()];
> delete item.getChild();
> item.setChild(nullptr);
> ```

Yeah, that works. Not sure it's necessary to set the child to be a nullptr because it looks like everything checks the child mask before accessing it, but agree that it's still perhaps safer.

> I envisioned the leaf level being a no-op; is there a measurable gain from swapping out the buffer?

Well it depends on the tree configuration. With standard configuration, you could in theory have only 4096 leaf nodes in your tree all of which hang off a single internal node that would otherwise be deallocated from a single thread, right? Even in more typical scenarios, you'll likely still end with a long tail as the grain size is so high. Better to maximize the parallelization by doing a parallel leaf-level pass to drop the buffers first I think.